src/wallet/init.cpp

```diff
@@ -324,7 +325,12 @@ bool WalletInit::Open() const
 void WalletInit::Start(CScheduler& scheduler) const
 {
     for (CWallet* pwallet : GetWallets()) {
-        pwallet->postInitProcess(scheduler);
+        pwallet->postInitProcess();
+    }
+
+    // Run a thread to flush wallet periodically
+    if (!CWallet::fFlushScheduled.exchange(true)) {
```

Comment:
`WalletInitInterface::Start()` is called only once, remove this check (and fFlushScheduled)?